utils/fasm/src/fasm_utils.h

```diff
@@ -22,6 +23,15 @@ std::vector<std::string> split_fasm_entry(std::string entry,
                                                  std::string delims,
                                                  std::string ignore);
 
+// Substitutes tags found in a string with their values provided by the map.
```

Comment:
Would be good to get some unit tests around this function, as it is a good candidate for it.